### PR TITLE
Run on_launch fairings after effective port is known in config.

### PR DESCRIPTION
**Pull request by toxeus**

### CODE REVIEW

**Review by jebrosen — CHANGES_REQUESTED** [review]
I believe this is a regression introduced at some point in the configuration revamp (<https://github.com/SergioBenitez/Rocket/commit/1fb061496d0234f84495723e6c95590064778785#diff-f5b44e8d22d6c82ac6288d0d2b08bf8a025e9822ea030c381ce206fae8e4c578R511>). The fix looks okay to me, just a small change requested in the test.

File: core/lib/tests/on_launch_fairing_can_inspect_port.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+extern crate rocket;
+
+use rocket::config::Config;
+use rocket::fairing::AdHoc;
+use rocket::figment::Figment;
+use std::sync::mpsc::sync_channel;
+
+#[rocket::async_test]
+async fn on_launch_fairing_can_inspect_port() {
+    let (tx, rx) = sync_channel::<u16>(0);
```
[the remark this answers is not in the document]
Comment (toxeus) [reply core/lib/tests/on_launch_fairing_can_inspect_port.rs]: Fixed

**Review by jebrosen — APPROVED** [review]
This might be the first test that actually tries to listen on a port. The port is set to `0`, so this seems okay to me.

An alternative approach may be to use #1154, which would let us test this functionality another way (but not this particular test).

**Comment by toxeus** [comment]
Ping @SergioBenitez

**Comment by SergioBenitez** [comment]
@toxeus Please allow collaborators to push to your branch next time!

**Comment by SergioBenitez** [comment]
Merged in 48fd83a31de0988e480678be9934f59d0df09f86.

**Pull request closed without merging**